terratorch/tasks/regression_tasks.py

```diff
@@ -115,6 +117,285 @@ def plot(self, val: Tensor | Sequence[Tensor] | None = None, ax=None):
     def reset(self) -> None:
         self.metric.reset()
 
+# TODO Maybe we could rewrite this method to inherit from PixelwiseRegressionTask.
```

Comment:
I would avoid introducing such a tight coupling. Maybe some of the common logic could be extracted into components which both classes have as attributes

Reply:
Right.